src/System.Buffers/src/System/Buffers/BufferPool.cs

```diff
@@ -41,7 +36,8 @@ public T[] RentBuffer(int size, bool clearBuffer = false)
             }
             else
             {
-                buffer = _buckets[SelectBucketIndex(size)].Rent();
+                int index = SelectBucketIndex(size);
+                buffer = LazyInitializer.EnsureInitialized(ref _buckets[index], () => new BufferBucket<T>(GetMaxSizeForBucket(index), NUMBER_OF_BUFFERS_IN_BUCKET)).Rent();
```

Comment:
What specifically do you believe should be fixed, and how?

I do think we should have another overload of EnsureInitialized that allows you to pass a TState through from the caller into the delegate.